grr/server/grr_response_server/gui/ui/components/client_details/client_details.ts

```diff
@@ -33,6 +102,116 @@ export class ClientDetails implements OnInit, OnDestroy {
     });
   }
 
+  getAccordionButtonState(
+      totalNumElements: number, currentMaxNumElementsShown: number,
+      initialMaxNumElementsShown: number): string {
+    if (totalNumElements > currentMaxNumElementsShown) {
+      return 'show-more';
+    } else if (totalNumElements <= initialMaxNumElementsShown) {
+      return 'no-button';
+    }
+    return 'show-less';
+  }
+
+  /**
+   * Returns the number of entries changed and a list of the aggregated changes
+   * descriptions (e.g. "2 users added")
+   */
+  static getDiffDescriptions(diff?: Diff<Client>[]): [number, string[]] {
```

Comment:
Please extract the diffing code out of the client details component into a standalone file.